File: C/Matrizes/Diagonal_negativos.c
#include <stdio.h>
#include <string.h>



int main ()
{
    int N;


    printf ("Qual a ordem da matriz? ");
    scanf ("%d", &N);

    int mat[N][N];

    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < N; j++)
        {
            printf ("Elemento [%d,%d]: ", i, j);
            scanf ("%d", &mat[i][j]);
        }
    }

    printf ("DIAGONAL PRINCIPAL:\n ");

    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < N; j++)
        {
            if (i == j)
            {
                printf("%d ", mat[i][j]);
            }
        }
    }

    int cont = 0;

    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < N; j++)
        {
            if (mat[i][j] < 0)
            {
                cont = cont + 1;

            }
        }
    }
    printf("\nQUANTIDADE DE NEGATIVOS: %d ", cont);



    return 0;
}

File: Python/Matrizes/Diagonal_negativos.py
M =int(input("Qual a ordem da matriz? "))


mat = [[0 for x in range(M)] for x in range(M)]

for i in range(0, M):
    for j in range(0, M):
        mat[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("DIAGONAL PRINCIPAL:")

for i in range(0, M):
    for j in range(0, M):
        if i == j:
            print(f"{mat[i][j]} ", end=" ")

negativos = 0

for i in range(0, M):
    for j in range(0, M):
        if mat[i][j] < 0:
            negativos = negativos + 1

print(f"\nQUANTIDADE DE NEGATIVOS = {negativos}")